lib/src/perf_benchmark_base.dart

```diff
@@ -0,0 +1,146 @@
+    perfControlAck = '${fifoDir.path}/perf_control_ack';
+
+    try {
+      final fifoResult = await Process.run('mkfifo', [perfControlFifo]);
+      if (fifoResult.exitCode != 0) {
+        throw ProcessException('mkfifo', [perfControlFifo],
+            'Cannot create fifo: ${fifoResult.stderr}', fifoResult.exitCode);
+      }
+      final ackResult = await Process.run('mkfifo', [perfControlAck]);
+      if (ackResult.exitCode != 0) {
+        throw ProcessException('mkfifo', [perfControlAck],
+            'Cannot create fifo: ${ackResult.stderr}', ackResult.exitCode);
+      }
+    } catch (e) {
+      await fifoDir.delete(recursive: true);
+      rethrow;
+    }
+  }
+
+  Future<void> _startPerfStat() async {
+    await _createFifos();
+    try {
+      perfProcess = await Process.start('perf', [
+        'stat',
+        '--delay',
+        '-1',
+        '--control',
+        'fifo:$perfControlFifo,$perfControlAck',
+        '-j',
+        '-p',
+        '$pid'
+      ]);
+      //await Future<void>.delayed(const Duration(seconds: 2));
+
+      openedFifo = File(perfControlFifo).openSync(mode: FileMode.writeOnly);
+
+      openedAck = File(perfControlAck).openSync();
+      openedFifo.writeStringSync('enable\n');
+      _waitForAck();
+    } catch (e) {
+      await fifoDir.delete(recursive: true);
+      rethrow;
+    }
+  }
+
+  Future<void> _stopPerfStat(int totalIterations) async {
+    try {
+      openedFifo.writeStringSync('disable\n');
+      openedFifo.closeSync();
+      _waitForAck();
+      openedAck.closeSync();
+      perfProcess.kill(ProcessSignal.sigint);
+      final lines =
+          utf8.decoder.bind(perfProcess.stderr).transform(const LineSplitter());
+      // Exit code from perf is -2 when terminated with SIGINT.
+      final exitCode = await perfProcess.exitCode;
+      if (exitCode != 0 && exitCode != -2) {
+        throw ProcessException(
+            'perf',
+            [
+              'stat',
+              '--delay',
+              '-1',
+              '--control',
+              'fifo:$perfControlFifo,$perfControlAck',
+              '-j',
+              '-p',
+              '$pid'
+            ],
+            (await lines.toList()).join('\n'),
+            exitCode);
+      }
+      final events = [
+        await for (final line in lines)
+          if (line.startsWith('{"counter-value" : '))
+            jsonDecode(line) as Map<String, dynamic>
+      ];
+      _reportPerfStats(events, totalIterations);
+    } finally {
+      await fifoDir.delete(recursive: true);
+    }
+  }
+
+  /// Measures the score for the benchmark and returns it.
+  Future<double> measurePerf() async {
+    setup();
+    // Warmup for at least 100ms. Discard result.
+    measureForImpl(warmup, 100);
+    await _startPerfStat();
+    // Run the benchmark for at least 2000ms.
+    var result = measureForImpl(exercise, minimumMeasureDurationMillis);
+    await _stopPerfStat(result.totalIterations);
+    teardown();
```

Comment:
teardown should be in a finally. I'm planning on fixing that in the other harness.

Reply:
Let's fix this at the same time it is fixed in the other harness.